backends/ebpf/bpfinclude/ebpf_registry.c

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include <stdio.h>
+#include "ebpf_registry.h"
+
+/**
+ * @brief Defines the structure of the central register.
+ * @details Defines a register type, which is a hashmap of
+ * tables identified by their name. Also associates a unique
+ * integer value as descriptor.
+ */
+typedef struct {
+    char name[VAR_SIZE];         // name of the map
+    int map_fd;                  // id of the map
+    struct bpf_map_def *table;   // ptr to the map
+    UT_hash_handle hh;           // makes this structure hashable
+} table_register;
+
+/**
+ * @brief A mapping between names and file descriptors.
+ * @details Maps the table names to their respective file descriptors.
+ * Provides a mechanism to find a table name by its integer value
+ * representation.
+ */
+typedef struct {
+    int map_fd;                 // file descriptor of the map
+    char name[VAR_SIZE];        // name of the map
+    UT_hash_handle hh;          // makes this structure hashable
+} fd_mapping;
+
+
+static int map_indexer = 0;
+
+/* Instantiation of the central register */
+table_register *reg_maps = NULL;
+fd_mapping *map_fds = NULL;
+
+
+table_register *_find_table(const char *name) {
```

Comment:
make this static and remove the underscore.

Reply:
done